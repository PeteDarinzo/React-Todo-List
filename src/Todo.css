
.Todo-complete {
  text-decoration-line: line-through;
}