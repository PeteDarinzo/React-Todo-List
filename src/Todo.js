import React from "react";
import "./Todo.css";


const Todo = ({ task, id, completed, toggleComplete, deleteTask }) => {

  const handleDeleteTask = (e) => {
    e.preventDefault();
    // deleteTask(e.target.parentElement.id);
    deleteTask(id);
  }

  const markComplete = (e) => {
    e.preventDefault();
    toggleComplete(id);
  }

  return (
    <div className="Todo" id={id} >

      <div className="Todo-task-container">
        <span className={completed ? "Todo-task Todo-complete" : "Todo-task"}><i class="fas fa-angle-double-right"></i> {task}</span>
      </div>

      <div className="Todo-buttons">

        <button className="Todo-delete" onClick={handleDeleteTask}>X</button>

        <button className="Todo-markComplete" onClick={markComplete}>Mark Complete</button>

      </div>

    </div>
  );
}

export default Todo;