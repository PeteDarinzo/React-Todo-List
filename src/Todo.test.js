import React from "react";
import { render } from '@testing-library/react';
import Todo from './Todo';



it("renders a todo without crashing", () => {
  <Todo
    task="walk dog"
    id={12345}
    completed={false}
    toggleComplete={() => null}
    deleteTask={() => null}
  />
});

it("matches the todo snapshot", () => {
  const { asFragment } = render(<Todo
    task="walk dog"
    id={12345}
    completed={false}
    toggleComplete={() => null}
    deleteTask={() => null}
  />
  );
  expect(asFragment()).toMatchSnapshot();
});