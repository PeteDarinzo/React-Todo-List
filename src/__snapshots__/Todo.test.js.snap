// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the todo snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="12345"
  >
    <div
      class="Todo-task-container"
    >
      <span
        class="Todo-task"
      >
        <i
          class="fas fa-angle-double-right"
        />
         walk dog
      </span>
    </div>
    <div
      class="Todo-buttons"
    >
      <button
        class="Todo-delete"
      >
        X
      </button>
      <button
        class="Todo-markComplete"
      >
        Mark Complete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
